src/thunderbots/software/network_input/filter/robot_filter.h

```diff
@@ -12,26 +12,65 @@
  * so we can make this module more generic and abstract away
  * the protobuf for testing
  */
-typedef struct
+typedef struct SSLRobotData
```

Comment:
Not sure why we've added `SSLRobotData` here? I'm not sure but I feel this could cause some subtle issues. Should be enough to do something like:

```
struct SSLRobotData {
...
};
```
and similarly for `FilteredRobotData` below?

This is out-of-scope, so feel free to leave it as it was (ie. without the extra `SSLRobotData` on this line, and similarly below).

Reply:
Without that name, the struct becomes an anonymous struct (technically has no name. The typdef doesn't count), which means it can't define a constructor. We need to define the constructor because the `Timestamp` member has no default constructor.

As far as I'm aware this won't cause issues since the struct name is only in scope within the `struct` tag, so is only accessible as `struct FilteredRobotData`. However I think it would be good to differentiate the names, and I've seen this as convention before.

To differentiate the struct name from the `typedef` I can prepend an underscore on struct name, aka `_FilteredRobotData`. This is roughly convention with C.

Reply:
Yeah, if we're following C conventions here I believe it would be `FilteredRobotData_t`?

Reply:
Appended `_t` to differentiate struct names